n_helpers.c

```diff
@@ -354,21 +407,24 @@ uint32_t NoteBinaryMaxEncodedLength(uint32_t unencodedLength)
          entire buffer use `NoteBinaryDataEncodedLength()` instead.
  */
 /**************************************************************************/
-const char * NoteBinaryReceive(uint8_t * buffer, uint32_t bufLen,
-                               uint32_t decodedOffset, uint32_t * decodedLen)
+const char * NoteBinaryReceive(uint8_t *buffer, uint32_t bufLen,
+                               uint32_t decodedOffset, uint32_t decodedLen)
 {
     // Validate parameter(s)
     if (!buffer) {
-        NOTE_C_LOG_ERROR("NULL buffer");
-        return ERRSTR("NULL buffer", c_err);
+        const char *err = ERRSTR("NULL buffer", c_bad);
+        NOTE_C_LOG_ERROR(err);
+        return err;
     }
-    if (!decodedLen) {
-        NOTE_C_LOG_ERROR("decodedLen cannot be NULL");
-        return ERRSTR("decodedLen cannot be NULL", c_err);
+    if (bufLen < (cobsEncodedMaxLength(decodedLen) + 1)) {
```

Comment:
I think that when we want to know about the required size of the buffer we should use `NoteBinaryMaxEncodedLength()` since that's what we expect the callers to use. 

Currently `NoteBinaryMaxEncodedLength` doesn't do the +1 so this call fails at present.

I thought we'd agreed on this last week that it's ok to call NoteBinaryMaxEncodedLength rather than replicating the code, especially when it's about validating buffer sizes.

(This is the 3rd time this has bit us btw.)

The cobsEncodedMaxLength() etc functions are suitable when talking about purely cobs conversion, and shouldn't make adjustments for the framing overhead.

Reply:
I don't understand how the code you're referencing (line 419) failed; it incorporates the `+1`.

Assuming what is currently sitting on line 419 works, it will be resolved by the underlying change to `cobsEncodedMaxLength()`.

Reply:
NoteBinaryMaxEncodedLength didn't add the +1, so it was returning 1 byte less. I feel adding +1 to cobsXXX functions is mixing concerns.

Reply:
I understand the concern, but I don't have the freedom to modify `cobsGuaranteedFit()`.

`cobsEncodedMaxLength()` was designed to be an analog to `cobsGuaranteedFit()`, and I believe it needs to follow the existing precedent.